examples/custom-usage-counter/custom_usage_counter.py

```diff
@@ -0,0 +1,112 @@
+
+
+@dataclass
+class CustomCounterServicePayload(Payload):
+    runtime: str = constraint(inf.INF_RUNTIME_NAME, default="test-counters")
+
+
+class CustomCounterService(Service):
+    @staticmethod
+    async def get_payload():
+        return CustomCounterServicePayload()
+
+    def __init__(self, *args, running_time_sec, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._running_time_sec = running_time_sec
+
+    async def start(self):
+        self._ctx.run("sleep", "1")
+        yield self._ctx.commit()
+
+    async def run(self):
+        start_time = datetime.now()
+        print(f"service {self.id} running on '{self.provider_name}'...")
+        while datetime.now() < start_time + timedelta(seconds=self._running_time_sec):
+            self._ctx.run("sleep", "1000")
+            yield self._ctx.commit()
+            usage: ActivityUsage = await self._ctx.get_usage()
+            cost = await self._ctx.get_cost()
+            print(f"total cost so far: {cost}; activity usage: {usage.current_usage}")
+            await asyncio.sleep(3)
+
+    async def shutdown(self):
+        async for s in super().shutdown():
+            yield s
+        print(f"service {self.id} stopped on '{self.provider_name}'")
+
+
+async def main(running_time_sec, subnet_tag, driver=None, network=None):
+
+    strategy = LeastExpensiveLinearPayuMS(
+        max_price_for={
+            com.Counter.CPU.value: Decimal("0.2"),
+            com.Counter.TIME.value: Decimal("0.1"),
+            "golem.usage.custom.counter": Decimal("0.1"),
+        }
+    )
+    strategy = DecreaseScoreForUnconfirmedAgreement(strategy, 0.5)
+
+    async with Golem(
+        budget=10.0, subnet_tag=subnet_tag, driver=driver, network=network, strategy=strategy
+    ) as golem:
+        instance_params = [{"running_time_sec": running_time_sec}]
+        cluster = await golem.run_service(CustomCounterService, instance_params=instance_params)
+
+        def print_instances():
+            print(f"instances: {[{s.id, s.state.value} for s in cluster.instances]}")
+
+        was_running = False
+
+        while True:
+            await asyncio.sleep(3)
+
+            if was_running and all([s.state == ServiceState.terminated for s in cluster.instances]):
+                print_instances()
+                print("All services were successfully terminated")
+                break
+
+            if len(cluster.instances) > 0:
+                print_instances()
+                was_running = True
+
+
+parser = build_parser("Custom Usage Counter Example")
+parser.add_argument(
+    "--running-time",
+    default=30,
+    type=int,
+    help="How long should the the service run (in seconds, default: %(default)s)",
+)
+args = parser.parse_args()
+enable_default_logger(
+    log_file=args.log_file,
+    debug_activity_api=True,
+    debug_market_api=True,
+    debug_payment_api=True,
+)
+asyncio.run(main(args.running_time, args.subnet_tag, args.driver, args.network))
```

Comment:
Is there any reason why we don't do `if __name__ == '__main__'` here?
Also, what with all this try/except, `task.cancel()` etc that we have in most examples?

I think this might be quite confusing for the reader who already knows & understands previous examples.

Reply:
Is this necessary if we are not importing this file? It's just an example that is meant to be run directly, it even starts with `#!/usr/bin/env python3`. I wanted this example to be as simple as possible, handling `NoPaymentAccountError` and `KeyboardInterrupt` would make it more complicated. It's not great when most of the code in the example is boilerplate.

Reply:
Someone might want to import `main` from this example and use it in some other way. Why forbid them? It might make "playing with an example" easier.
I think this `if __name__ ...` is just a good standard, although I won't insist.

About NoPaymentAccountError/KeyboardInterrupt etc: I definitely agree this should be simplified, but I'd rather have the same in *all* examples (except hello world) and then fix it in *all* examples at the same time. But I won't insist here either : )

Reply:
I added `if __name__ == "__main__":` (https://github.com/golemfactory/yapapi/pull/611/commits/9b238e6e8a24e0423ccd08d966e04165d36b4c5b)

BTW: this example does not run indefinitely, so Ctrl+C wouldn't be typically pressed:
```py
while datetime.now() < start_time + timedelta(seconds=self._running_time_sec):
...
parser.add_argument(
    "--running-time",
    default=30,
```